subsys/bluetooth/controller/ll_sw/lll.h

```diff
@@ -155,18 +155,26 @@ struct ull_hdr {
 struct lll_hdr {
 	void *parent;
 	uint8_t is_stop:1;
+#if defined(CONFIG_BT_CTLR_NEXT_GEN_SCHEDULING)
+	uint8_t score;
```

Comment:
I will separate the priority calculation and scoring in a separate commit. Here are the basics:

_Scoring_ is used as a term for "earning priority", meaning the higher score an event gets, the more likely it is that it will get to go in the air the next time it's scheduled ("event" here used as the recurring scheduling of an instance of a certain role, e.g. ADV or CONN[0]).

An increase in score is intended to happen whenever an event prepare is either scheduled and pre-empted before it ended, or if it was queued in the prepare pipeline and when popped, was too late to execute (referred to as "aborted" and "too_late").

So when two events "compete" for the same air-time, loser will gain a score of 1 each time the other gets the air, and thus gain higher priority (using its score) the next time, thereby winning the air-time.
This is similar to how the `lazy_current` used to increment when ticker priority resolve caused a loss. Latency is also maintained, in order to be able to supply the correct value in the prepare callback.